### ISSUE TITLE
test crash on freebsd 

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#844]
Testing the borg binary runs into an issue on freebsd:

```
==> freebsd64: ___________________ ArchiverTestCaseBinary.test_delete_repo ____________________
==> freebsd64: self = <borg.testsuite.archiver.ArchiverTestCaseBinary testMethod=test_delete_repo>
==> freebsd64:     def test_delete_repo(self):
==> freebsd64:         self.create_regular_file('file1', size=1024 * 80)
==> freebsd64:         self.create_regular_file('dir2/file2', size=1024 * 80)
==> freebsd64:         self.cmd('init', self.repository_location)
==> freebsd64:         self.cmd('create', self.repository_location + '::test', 'input')
==> freebsd64:         self.cmd('create', self.repository_location + '::test.2', 'input')
==> freebsd64:         os.environ['BORG_DELETE_I_KNOW_WHAT_I_AM_DOING'] = 'no'
==> freebsd64:         self.cmd('delete', self.repository_location, exit_code=2)
==> freebsd64: >       self.archiver.exit_code = 0
==> freebsd64: E       AttributeError: 'NoneType' object has no attribute 'exit_code'
==> freebsd64: py34/lib/python3.4/site-packages/borg/testsuite/archiver.py:736: AttributeError
==> freebsd64: ======= 1 failed, 441 passed, 112 skipped, 3 xpassed in 2917.08 seconds ========
```

**Comment by jungle-boogie** [comment]
> Testing the borg binary runs into an issue on freebsd:

I downloaded the binary and tested this on freebsd 10.2 and there was no problem. 

```
./borg-freebsd64
usage: borg-freebsd64 [-h] [-V] <command> ...

Borg - Deduplicated Backups

optional arguments:
  -h, --help            show this help message and exit
  -V, --version         show version number and exit

```

How did you have this issue?

**Comment by enkore** [comment]
My fault, I didn't take the BinaryTestCase into account. For the normal in-process test cases self.archiver is the Archiver instance which lives between invocations of self.cmd(). Hotfix: set exit_code not only in `__init__` of Archiver, but also when parsing a command line / before invoking a do_XXX?

**Comment by ThomasWaldmann** [comment]
@jungle-boogie I've seen it when starting the vagrant freebsd test machine.

**Comment by ThomasWaldmann** [comment]
fixed in 1.0-maint, soon to be merged into master.
